src/datasets/load.py

```diff
@@ -512,7 +514,7 @@ def infer_module_for_data_files(
         raise ValueError(f"Couldn't infer the same data file format for all splits. Got {split_modules}")
     if not module_name:
         path = f" in {path}. " if path else ". "
-        raise FileNotFoundError(f"No (supported) data files or dataset script found{path}")
+        raise DataFilesNotFoundError(f"No (supported) data files or dataset script found{path}")
```

Comment:
Maybe `NotADatasetError` would be a better name for this error (the "dataset script" is not a data file)

Reply:
I had kept the original message errors for backward compatibility: before we were raising the same `FileNotFoundError` for all cases and end users could only differentiate among them by parsing the error messages
- This is indeed the case for datasets-server; see https://github.com/huggingface/datasets/issues/6397#issue-1987622152
  > To differentiate, we must parse the error message (only the end is different). We should have a different exception for these two errors.

However, I would propose changing the error message in this specific case (and keep the error name) because:
- this error is raised by a function that just checks the data files and tries to infer the module from them
- for the cases where it is a script-dataset, this is handled in other part of the code inside `dataset_module_factory`
- anyway, we are pushing no-script datasets, so their usage should be exceptional

Reply:
Done.